cdap-ui/app/hydrator/services/hydrator-service.js

```diff
@@ -434,6 +434,15 @@ class HydratorPlusPlusHydratorService {
       userRuntimeArgumentsMap
     };
   }
+
+  getMacrosWithNonEmptyValues(macrosMap) {
+    let macrosMapCopy = Object.assign({}, macrosMap);
+    let {keysWithMissingValue} = this.myHelpers.objHasMissingValues(macrosMapCopy);
+    keysWithMissingValue.forEach(key => {
```

Comment:
I hae modified `objHasMissingValues` to always return an object with keys `res` and `keysWithMissingValue`, so if `res` is `false` then `keysWithMissingValue` returns an empty array. So far this is the only place that uses that helper function anyways, and it didn't make sense to have two different return formats.